src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -41,24 +45,74 @@ public LogicManager(Model model, Network network) {
         history = new CommandHistory();
         bookShelfParser = new BookShelfParser(model.getAliasList());
         undoStack = new UndoStack();
+        key = model.getKey();
+        if (key == null) {
+            isEncrypt = false;
+        } else {
+            isEncrypt = true;
+        }
     }
-
+    //@@author 592363789
+    /**
+     * Constructor for test use.
+     */
+    public LogicManager(Model model, Network network, boolean isEncrypt) {
```

Comment:
Should go into some `TestUtil` class

Reply:
What I should do?

Reply:
Can put inside TestApp similar to getModel. Depends on what it's for??

Reply:
I cannot figure out how to solve two systemtest fail, can you help me check how to solve? Thank you!